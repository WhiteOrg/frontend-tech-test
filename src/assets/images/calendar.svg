<svg xmlns="http://www.w3.org/2000/svg" width="18" height="20" fill="none" viewBox="0 0 18 20">
    <path fill="#fff" fill-rule="evenodd" d="M16 2h-1V0h-2v2H5V0H3v2H2C.89 2 .01 2.9.01 4L0 18c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM4 9v2h2V9H4zm6 0H8v2h2V9zm4 0v2h-2V9h2zM2 18h14V7H2v11z" clip-rule="evenodd"/>
</svg>
