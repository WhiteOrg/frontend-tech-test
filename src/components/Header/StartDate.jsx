import React from 'react'
// 
import CalendarIcon from '../../svgComponents/CalendarIcon'
import styled from 'styled-components'



const Container = styled.div`
    display: flex;
    margin: 2em 0;
    align-items: center;
`

const DateText = styled.p`
    font-size: 0.8em;
    color: #fff;
    margin-left: 1em;
`

// JSX
export default function StartDate({ date }) {
    return(
        <Container>
            <CalendarIcon />
            <DateText>Starts on {date} </DateText>
        </Container>
    )
}