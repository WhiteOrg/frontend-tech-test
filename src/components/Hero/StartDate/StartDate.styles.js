import styled from "styled-components";

export const Container = styled.div`
  color: #fff;
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: 1rem;
  font-weight: 700;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 1.5rem 0;
  @media ${({ theme }) => theme.mediaQueries["tablet"]} {
    font-size: 1.1rem;
  }
  @media ${({ theme }) => theme.mediaQueries["laptop"]} {
    font-size: 1.3rem;
  }
  @media ${({ theme }) => theme.mediaQueries["desktop"]} {
    font-size: 1.5rem;
  } ;
`;

export const Span = styled.div`
  margin-left: 0.8rem;
`;
