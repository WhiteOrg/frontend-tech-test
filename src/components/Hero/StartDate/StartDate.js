import React from "react";
import { Container, Span } from "./StartDate.styles";

const StartDate = ({ startDate, calendarImg }) => {
  return (
    <Container>
      <span>
        {calendarImg ? <img src={calendarImg} alt="calendar" /> : null}{" "}
      </span>

      <Span>Starts on {startDate}</Span>
    </Container>
  );
};

export default StartDate;
