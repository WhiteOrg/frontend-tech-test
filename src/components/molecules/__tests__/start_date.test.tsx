import React from "react";
import renderer from "react-test-renderer";
import StartDate from "../start_date";
import calendar from "../../../images/calendar.svg";

describe("MOLECULES: Render testing", () => {
  test("Should render StartDate", () => {
    const component = renderer.create(
      <div>
        <StartDate icon={calendar} startDate="2020-08-06T10:06:29.578Z" />
      </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
