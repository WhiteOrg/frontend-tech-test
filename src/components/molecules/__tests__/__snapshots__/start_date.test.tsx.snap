// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATOMS: Render testing Should render StartDate 1`] = `
<div>
  <div
    className="sc-AxirZ ehRAKx"
  >
    <img
      className="sc-AxiKw bOTXee"
      src="calendar.svg"
    />
    <p
      className="sc-AxjAm jlKNmI"
      color="white"
      size="small"
    >
      Starts On 
      Thu Aug 06 2020
      ,
       
      11:06:29 AM
    </p>
  </div>
</div>
`;

exports[`MOLECULES: Render testing Should render StartDate 1`] = `
<div>
  <div
    className="sc-AxirZ ehRAKx"
  >
    <img
      className="sc-AxiKw bOTXee"
      src="calendar.svg"
    />
    <p
      className="sc-AxjAm jlKNmI"
      color="white"
      size="small"
    >
      Starts On 
      Thu Aug 06 2020
      ,
       
      11:06:29 AM
    </p>
  </div>
</div>
`;
